### ISSUE TITLE
-Wsometimes-uninitialized in fs/gfs2/bmap.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1951]
```
fs/gfs2/bmap.c:891:9: error: variable 'ret' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
  891 |                                 if (flags & IOMAP_REPORT) {
      |                                     ^~~~~~~~~~~~~~~~~~~~
fs/gfs2/bmap.c:948:9: note: uninitialized use occurs here
  948 |         return ret;
      |                ^~~
fs/gfs2/bmap.c:891:5: note: remove the 'if' if its condition is always true
  891 |                                 if (flags & IOMAP_REPORT) {
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~
  892 |                                         ret = -ENOENT;
  893 |                                         goto unlock;
  894 |                                 } else {
      |                                   ~~~~~~
  895 |                                         iomap->offset = pos;
      |                                         ~~~~~~~~~~~~~~~~~~~~
  896 |                                         iomap->length = length;
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~
  897 |                                         goto hole_found;
      |                                         ~~~~~~~~~~~~~~~~
  898 |                                 }
      |                                 ~
fs/gfs2/bmap.c:890:8: error: variable 'ret' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
  890 |                         if (pos >= size) {
      |                             ^~~~~~~~~~~
fs/gfs2/bmap.c:948:9: note: uninitialized use occurs here
  948 |         return ret;
      |                ^~~
fs/gfs2/bmap.c:890:4: note: remove the 'if' if its condition is always true
  890 |                         if (pos >= size) {
      |                         ^~~~~~~~~~~~~~~~
fs/gfs2/bmap.c:886:8: error: variable 'ret' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
  886 |                         if (pos + length > max_size)
      |                             ^~~~~~~~~~~~~~~~~~~~~~~
fs/gfs2/bmap.c:948:9: note: uninitialized use occurs here
  948 |         return ret;
      |                ^~~
fs/gfs2/bmap.c:886:4: note: remove the 'if' if its condition is always true
  886 |                         if (pos + length > max_size)
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
  887 |                                 goto unstuff;
fs/gfs2/bmap.c:869:9: note: initialize the variable 'ret' to silence this warning
  869 |         int ret;
      |                ^
      |                 = 0
3 errors generated.
```

There is [a kernel test robot email](https://lore.kernel.org/202310242105.0fHmealv-lkp@intel.com/) with this reported to the author. If there is no movement on it in a day or so, I will ping it for direct input due to build breakage with `-Werror` / `CONFIG_WERROR=y`.

**Comment by nathanchance** [comment]
I can see this is now fixed in the gfs2 tree: https://git.kernel.org/gfs2/linux-gfs2/c/81118680a9cf6b24ea7943e0a7b2a98bf8d2b676

**Comment by nathanchance** [comment]
This is now fixed in next-20231026.
